staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -473,18 +473,25 @@ func (config *inClusterClientConfig) ClientConfig() (*restclient.Config, error)
 		return nil, err
 	}
 
-	// in-cluster configs only takes a host, token, or CA file
-	// if any of them were individually provided, ovewrite anything else
+	// in-cluster configs only takes --as, --as-group, --request-timeout
```

Comment:
Removing token break e2e test here:
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/kubectl/kubectl.go#L684
Override icc.Host doesn't make sense.
But override BearerToken and CAFile makes sense.
I will took all them back.